crates/shared/src/price_estimation/native/oneinch.rs

```diff
@@ -0,0 +1,237 @@
+use {
+    super::{NativePriceEstimateResult, NativePriceEstimating},
+    crate::price_estimation::{
+        Estimate,
+        PriceEstimateResult,
+        PriceEstimating,
+        PriceEstimationError,
+        Query,
+    },
+    anyhow::{anyhow, Result},
+    futures::{future::BoxFuture, FutureExt},
+    model::order::OrderKind,
+    number::nonzero::U256 as NonZeroU256,
+    primitive_types::{H160, U256},
+    reqwest::Client,
+    std::{
+        collections::HashMap,
+        sync::{Arc, Mutex},
+        time::Duration,
+    },
+};
+
+const BASE_URL: &str = "https://api.1inch.dev/";
+const REFRESH_INTERVAL: Duration = Duration::from_secs(12);
+
+struct OneInch {
+    client: Client,
+    chain_id: u64,
+    // Denominated in wei
+    prices: Arc<Mutex<HashMap<H160, U256>>>,
+    native_price_estimation_amount: NonZeroU256,
+    native_token: H160,
+}
```

Comment:
I think I would drop everything but the `prices` cache from this struct. Because all the requests are done by the background task the estimator only needs to know about the `prices` cache.
Then all the other data could be moved into a separate struct or simply be passed by value into the function that spawns the background task.
This would improve the separation of concerns IMO.